datafusion/physical-plan/src/common.rs

```diff
@@ -373,6 +375,38 @@ pub fn batch_byte_size(batch: &RecordBatch) -> usize {
     batch.get_array_memory_size()
 }
 
+/// Constructs the mapping between a projection's input and output
+pub fn calculate_projection_mapping(
```

Comment:
To make this easier to discover, perhaps it could be a method in `ProjectionMapping` such as `ProjectionMapping::try_new`

Reply:
For now `ProjectionMapping` is just a type alias, but we plan to promote it to a full-fledged type in a follow on PR -- I agree that it makes sense to convert this into a method then.

Reply:
FWIW I hacked up a version here and it worked great: https://github.com/apache/arrow-datafusion/pull/8033